pkgs/build-support/dotnet/make-nuget-deps/default.nix

```diff
@@ -1,14 +1,35 @@
-{ linkFarmFromDrvs, fetchurl }:
+{ linkFarmFromDrvs, fetchurl, stdenvNoCC, zip }:
 { name, nugetDeps ? import sourceFile, sourceFile ? null }:
 linkFarmFromDrvs "${name}-nuget-deps" (nugetDeps {
   fetchNuGet = { pname, version, sha256 ? "", hash ? ""
     , url ? "https://www.nuget.org/api/v2/package/${pname}/${version}" }:
-    fetchurl {
+    stdenvNoCC.mkDerivation {
       name = "${pname}.${version}.nupkg";
-      # There is no need to verify whether both sha256 and hash are
-      # valid here, because nuget-to-nix does not generate a deps.nix
-      # containing both.
-      inherit url sha256 hash;
+      src = fetchurl {
+        # There is no need to verify whether both sha256 and hash are
+        # valid here, because nuget-to-nix does not generate a deps.nix
+        # containing both.
+        inherit url sha256 hash;
+      };
+      dontUnpack = true;
+      nativeBuildInputs = [ zip ];
+      installPhase = "cp $src $out";
+      # NuGet.org edits packages by signing them during upload, which makes
+      # those packages nondeterministic depending on which source you
+      # get them from. We fix this by stripping out the signature file.
+      # Signing logic is https://github.com/NuGet/NuGet.Client/blob/128a5066b1438627ac69a2ffe9de564b2c09ee4d/src/NuGet.Core/NuGet.Packaging/Signing/Archive/SignedPackageArchiveIOUtility.cs#L518
+      # Non-NuGet.org sources might not have a signature file.
+      fixupPhase = ''
+        runHook preFixup
+        if zip --show-files "$out" | grep --quiet --line-regexp '[[:space:]]\+\.signature\.p7s'; then
```

Comment:
Instead of hacking up a pipeline to check whether the archive contains a signature file, can we instead check the documented `zip` exit code?
https://linux.die.net/man/1/zip#:~:text=12%0A%0Azip%20has%20nothing%20to%20do

```bash
zip --temp-path "$TMPDIR" --delete "$out" .signature.p7s || (( $? == 12 ))
```